ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_api_gnmi.schema.yml

```diff
@@ -0,0 +1,62 @@
+# yaml-language-server: $schema=../../../../plugins/plugin_utils/schema/avd_meta_schema.json
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  management_api_gnmi:
+    type: dict
+    keys:
+      provider:
+        type: str
+        default: eos-native
+      transport:
+        type: dict
+        keys:
+          grpc:
+            type: list
+            items:
+              type: dict
+              keys:
+                name:
+                  type: str
+                  description: Transport name
+                ssl_profile:
+                  type: str
+                  description: SSL profile name
+                vrf:
+                  type: str
+                  description: VRF name is optional
+                notification_timestamp:
+                  type: str
+                  valid_values: ["send-time", "last-change-time"]
+                  description: |
+                    Per the GNMI specification, the default timestamp field of a notification message is set to be
+                    the time at which the value of the underlying data source changes or when the reported event takes place.
+                    In order to facilitate integration in legacy environments oriented around polling style operations,
+                    an option to support overriding the timestamp field to the send-time is available from EOS 4.27.0F.
+                ip_access_group:
+                  type: str
+                  description: ACL name
+      enable_vrfs:
+        type: list
+        primary_key: name
+        convert_types:
+          - dict
+        description: |
+          Enable VRFs will be deprecated in AVD v4.0.
+          These should not be mixed with the new keys above.
+        items:
+          type: dict
+          keys:
+            name:
+              type: str
+              description: VRF name
+            access_group:
+              type: str
+              description: Standard IPv4 ACL name
+      octa:
+        type: dict
```

Comment:
Should not it be a boolean? - not sure how it would have been used before as our documentation is empty

```
{%     if management_api_gnmi.octa is defined %}
   provider eos-native
{%     endif %}
```

Though looking at molecule we are using it like this:

```
--management_api_gnmi:
    enable_vrfs:
      MGMT:
        access_group: ACL-GNMI
      MONITORING:
    octa:
```

so probably just the fact that the dict exist makes it work .. could it break existing implementations ? (we can always argue that it is how it was intended to be used ..)

@ClausHolbechArista

Reply:
This is one of the problematic data models which is not well defined. It should have been a boolean, but it was created like a "dangling" key, which would also work if set with `null`, so it is difficult to provide a schema covering all variations. Normally we have made these keys into dicts and provided an empty dict in the molecule vars, but you are correct that it might be breaking here.
Let us sync up offline and discuss a solution.

Reply:
So since the documented value is `null` and we basically ignore `null` values during schema validation, this will not block anything. I suggest to keep it as dict in the schema, and move on. This is deprecated for 4.0, so it will only be an issue if someone used another value than `null` or `dict` for 3.8+.